tests/test_lambda.py

```diff
@@ -374,3 +374,132 @@ def test_mark_and_match(self):
             after_tags,
             {'custodian_next': 'Resource does not meet policy: delete@2019/02/09',
              'xyz': 'abcdef'})
+
+
+
+class TestModifyVpcSecurityGroupsAction(BaseTest):
+
+    def test_lambda_remove_matched_security_groups(self):
+
+        # Test conditions:
+        #   - list with two functions, matching only one "resource-fixer"
+        #    - this function is in a VPC and has 3 SGs attached
+        #    - removing a third SG, "sg_controllers" (sg-c573e6b3)
+        #    - start with 3 SGs, end with 2, match function by regex
+
+        session_factory = self.replay_flight_data(
+            "test_lambda_remove_matched_security_groups"
+        )
+
+        p = self.load_policy(
+            {
+                "name": "lambda-remove-matched-security-groups",
+                "resource": "lambda",
+                "filters": [
+                    {
+                        "type": "value",
+                        "key": "FunctionName",
+                        "value": "resource-fixer",
+                        "op": "eq",
+                    },
+                    {
+                        "type": "security-group",
+                        "key": "GroupName",
+                        "value": ".*controllers",
+                        "op": "regex",
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "modify-security-groups",
+                        "remove": "matched",
+                        "isolation-group": "sg-01a19f602ecaf25f4",
+                    }
+                ],
+            },
+            session_factory=session_factory,
+        )
+        
+        clean_p = self.load_policy(
```

Comment:
So rather than using a policy to verify a policy, we generally strongly prefer to use an api client here as it will result in significantly less recorded data. 

client = session_factory().client(“lambda”)
client.get_function()

Reply:
changed these to use session_factory instead of a 2nd policy